endaq/calc/shock.py

```diff
@@ -17,29 +17,186 @@
+                scipy.signal.lfilter,
+                *_abs_accel_coeffs(omega, Q, T),
+                axis=0,
+                ),
+        raw=True,
+        )
+
+
+def _rel_velocity_coeffs(omega, Q, T):
+    """
+    Calculate the coefficients of the Z-domain transfer function for the
+    relative velocity according to ISO 18431-4
+
+    :param omega: the natural frequency of the system
+    :param Q: the quality factor of the system
+    :param T: the time step in seconds
+    :return: the coefficients of the Z-domain transfer function b, a
+
+    .. seealso::
+
+        - `ISO 18431-4 Mechanical vibration and shock — Signal processing — Part 4: Shock-response spectrum analysis`
+          Explicit implementations of digital filter coefficients for shock spectra.
+    """
+    A = omega*T/(2.*Q)
+    B = omega*T*np.sqrt(1. - 1./(4.*(Q**2.)))
+    C = np.exp(-A)*np.sin(B)/np.sqrt(4.*(Q**2.) - 1.)
+    D = T*(omega**2.)
+
+    b = (
+        (-1. + np.exp(-A)*np.cos(B) + C)/D,
+        (1. - np.exp(-2.*A) - 2.*C)/D,
+        (np.exp(-2.*A) - np.exp(-A)*np.cos(B) + C)/D,
+        )
+    a = (
+        1.,
+        -2.*np.exp(-A)*np.cos(B),
+        np.exp(-2.*A),
+        )
+
+    return b, a
+
+
+def rel_velocity(accel: pd.DataFrame, omega: float, damp: float = 0.0) -> pd.DataFrame:
+    """
+    Calculate the relative velocity for a SDOF system.
+
+    The "relative" displacement follows the transfer function:
+        H(s) = L{z'(t)}(s) / L{y"(t)}(s) = (1/s)(Z(s)/Y(s))
+    for the PDE:
         z" + (2ζω)z' + (ω²)z = -y"
 
+    :param accel: the absolute acceleration y"
+    :param omega: the natural frequency ω of the SDOF system
+    :param damp: the damping coefficient ζ of the SDOF system
+    :return: the relative displacement z of the SDOF system
+
     .. seealso::
 
         - `Pseudo Velocity Shock Spectrum Rules For Analysis Of Mechanical Shock, Howard A. Gaberson <https://info.endaq.com/hubfs/pvsrs_rules.pdf>`_
         - `SciPy transfer functions <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.TransferFunction.html>`_
           Documentation for the transfer function class used to characterize the
           relative displacement calculation.
+        - `SciPy biquad filter <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.lfilter.html>`__
+          Documentation for the biquad function used to implement the transfer
+          function.
+        - `ISO 18431-4 Mechanical vibration and shock — Signal processing — Part 4: Shock-response spectrum analysis`
+          Explicit implementations of digital filter coefficients for shock spectra.
+    """
+    T = utils.sample_spacing(accel)
+    Q = 1./(2.*damp)
+
+    return accel.apply(
+        functools.partial(
+                scipy.signal.lfilter,
+                *_rel_velocity_coeffs(omega, Q, T),
+                axis=0,
+                ),
+        raw=True,
+        )
+
+
+def _rel_displ_coeffs(omega, Q, T):
```

Comment:
abbreviated form: as-is 👍

non-abbreviated form:
```suggestion
def _relative_displacement_coeffs(omega, Q, T):
```